src/agent/coverage/src/sancov.rs

```diff
@@ -0,0 +1,429 @@
+        stop = $stop: ident,
+        ty = $ty: expr,
+    ) => {
+        define_table_getter!(name = $name, start = $start, stop = $stop, ty = $ty,);
+    };
+}
+
+impl SancovDelimiters {
+    /// Return the most compiler-specific Sancov inline counter or bool flag table, if any.
+    pub fn inline_table(&self, pad: bool) -> Option<SancovTable> {
+        // With MSVC, the LLVM delimiters are typically linked in alongside the
+        // MSVC-specific symbols. Check for MSVC-delimited tables first, though
+        // our validation of table size _should_ make this unnecessary.
+
+        if let Some(table) = self.msvc_bools_table(pad) {
+            return Some(table);
+        }
+
+        if let Some(table) = self.msvc_counters_table(pad) {
+            return Some(table);
+        }
+
+        if let Some(table) = self.msvc_preview_counters_table(pad) {
+            return Some(table);
+        }
+
+        // No MSVC tables found. Check for LLVM-emitted tables.
+
+        if let Some(table) = self.llvm_bools_table(pad) {
+            return Some(table);
+        }
+
+        if let Some(table) = self.llvm_counters_table(pad) {
+            return Some(table);
+        }
+
+        None
+    }
+
+    /// Return the most compiler-specific PC table, if any.
+    pub fn pcs_table(&self, pad: bool) -> Option<SancovTable> {
+        // Check for MSVC tables first.
+        if let Some(table) = self.msvc_pcs_table(pad) {
+            return Some(table);
+        }
+
+        if let Some(table) = self.llvm_pcs_table(pad) {
+            return Some(table);
+        }
+
+        None
+    }
+
+    define_table_getter!(
+        name = llvm_bools_table,
+        start = llvm_bools_start,
+        stop = llvm_bools_stop,
+        ty = SancovTableTy::Bools,
+    );
+
+    define_table_getter!(
+        name = llvm_counters_table,
+        start = llvm_counters_start,
+        stop = llvm_counters_stop,
+        ty = SancovTableTy::Counters,
+    );
+
+    define_table_getter!(
+        name = llvm_pcs_table,
+        start = llvm_pcs_start,
+        stop = llvm_pcs_stop,
+        ty = SancovTableTy::Pcs,
+    );
+
+    define_table_getter!(
+        name = msvc_bools_table,
+        start = msvc_bools_start,
+        stop = msvc_bools_stop,
+        ty = SancovTableTy::Bools,
+    );
+
+    define_table_getter!(
+        name = msvc_counters_table,
+        start = msvc_counters_start,
+        stop = msvc_counters_stop,
+        ty = SancovTableTy::Counters,
+    );
+
+    define_table_getter!(
+        name = msvc_pcs_table,
+        start = msvc_pcs_start,
+        stop = msvc_pcs_stop,
+        ty = SancovTableTy::Pcs,
+    );
+
+    define_table_getter!(
+        name = msvc_preview_counters_table,
+        start = msvc_preview_counters_start,
+        stop = msvc_preview_counters_stop,
+        ty = SancovTableTy::Counters,
+    );
+
+    pub fn insert(&mut self, delimiter: Delimiter, offset: u32) {
+        use Delimiter::*;
```

Comment:
Importing from an enum defined in this file is a unusual pattern that I don't think we do anywhere else.  I recognize this is copied from a previous commit, but seeing it here I found it odd.

Reply:
Changed. There are a small number of other places where we do it, but it's not super high value here.